File: src/Day1.kt
import java.io.File

fun main() {
    val newLine = System.lineSeparator()
    val inputFile = File("src/day1.txt")
    val caloriesPerElf = inputFile.readText().trim()
        .split("$newLine$newLine")
        .map { it.split(newLine).map(String::toInt).reduce{a,b -> a+b} }

    val mostCalories = caloriesPerElf.max()
    val top3Calories = caloriesPerElf.sorted().takeLast(3).sum()

    println(mostCalories)
    println(top3Calories)
}

File: src/Day1.py
import os

with open('src/day1.txt') as f:
    listOfElvesWithSnacks = f.read().split("\n\n")

snacksByElf = [elf.splitlines() for elf in listOfElvesWithSnacks]
snacksByElfInt = [list(map(int, elf)) for elf in snacksByElf]
summedCalByElf = [sum(elf) for elf in snacksByElfInt]
summedCalByElf.sort(reverse=True)

print(max(summedCalByElf))
print(sum(summedCalByElf[:4]))
